docs/en/engines/table-engines/mergetree-family/invertedindexes.md

```diff
@@ -60,70 +59,130 @@ ENGINE = MergeTree
 
-Being a type of skipping index, full-text indexes can be dropped or added to a column after table creation:
+Being a type of skipping index, text indexes can be dropped or added to a column after table creation:
 
 ```sql
 ALTER TABLE tab DROP INDEX inv_idx;
 ALTER TABLE tab ADD INDEX inv_idx(s) TYPE text(tokenizer = 'default');
 ```
 
-To use the index, no special functions or syntax are required. Typical string search predicates automatically leverage the index. As
-examples, consider:
+## Using a Text Index {#using-a-text-index}
+
+To use the index, no special functions or syntax are required.
+Typical string search predicates automatically leverage the index. As examples, consider:
 
 ```sql
-INSERT INTO tab(key, str) values (1, 'Hello World');
+INSERT INTO tab(key, str) VALUES (1, 'Hello World');
 SELECT * from tab WHERE str == 'Hello World';
 SELECT * from tab WHERE str IN ('Hello', 'World');
 SELECT * from tab WHERE str LIKE '%Hello%';
 SELECT * from tab WHERE multiSearchAny(str, ['Hello', 'World']);
 SELECT * from tab WHERE hasToken(str, 'Hello');
 ```
 
-The full-text index also works on columns of type `Array(String)`, `Array(FixedString)`, `Map(String)` and `Map(String)`.
+The text index also works on columns of type `Array(String)`, `Array(FixedString)`, `Map(String)` and `Map(String)`.
+
+Like for other secondary indices, each column part has its own text index.
+Furthermore, each text index is internally divided into "segments".
+The existence and size of the segments are generally transparent to users but the segment size determines the memory consumption during index construction (e.g. when two parts are merged).
+Configuration parameter `max_digestion_size_per_segment` (default: 256 MB) controls the amount of data read from the underlying column before a new segment is created.
+The default value of the parameter provides a good balance between memory usage and performance for most use cases.
+Incrementing it raises the intermediate memory consumption for index construction but also improves lookup performance since fewer segments need to be checked on average to evaluate a query.
+
+### Functions Support {#functions-support}
+
+The conditions in the `WHERE` clause contains calls of the functions that operate with columns.
+If the column is a part of an index, ClickHouse tries to use this index when performing the functions.
+ClickHouse supports different subsets of functions for the `text` index.
+
+#### equals and notEquals {#functions-example-equals-notequals}
+
+Functions `=` (equals) and `!=` (notEquals) check if the column contains rows which match the entire search term.
+Therefore, they work reasonably only with text indexes with `no_op` tokenizer.
+
+#### like, notLike and match {#functions-example-like-notlike-match}
+
+In order to use functions `like`, `notLike`, and `match` with the `text` index, the search term should be in a way that complete tokens can be extracted from it.
+
+Example:
+
+```sql
```

Comment:
Could we add a sentence or two how LIKE and friends work with non-standard tokenizers?

Reply:
Actually, only `default` and `ngram` tokenizers supports `like`, `notLike` and `match` functions. For non-standard tokenizers the server throws an exception, e.g. https://github.com/ClickHouse/ClickHouse/blob/46a9656b64e6b9edc8a54f51d41def1fc049b4c3/src/Interpreters/ITokenExtractor.cpp#L349

I updated the implementation to not throw, instead do not use the text index. Also, added a note section in docs.